plugins/kinetic-dp/fu-kinetic-dp-device.c

```diff
@@ -0,0 +1,270 @@
+		g_prefix_error(error, "device firmware update failed: ");
+		return FALSE;
+	}
+	fu_progress_step_done(progress);
+
+	/* success */
+	return TRUE;
+}
+
+/* making the connection to the pysical device */
+static gboolean
+fu_kinetic_dp_device_rescan(FuDevice *device, GError **error)
+{
+	FuKineticDpDevice *self = FU_KINETIC_DP_DEVICE(device);
+	g_autoptr(FuKineticDpConnection) connection = NULL;
+	g_autofree gchar *name = NULL;
+	g_autofree gchar *version = NULL;
+	g_autofree gchar *guid = NULL;
+	const gchar *chip_id;
+	guint8 buf_ver[16];
+	KtDpDevInfo *dp_dev_info = NULL;
+
+	connection = fu_kinetic_dp_connection_new(fu_udev_device_get_fd(FU_UDEV_DEVICE(self)));
+
+	/* TODO: now only support to do ISP for Host chip */
+	if (!fu_kinetic_dp_aux_isp_read_basic_device_info(self, DEV_HOST, &dp_dev_info, error)) {
+		g_prefix_error(error, "failed to read device info: ");
+		return FALSE;
+	}
+
+	g_debug("device scanning found branch_id_str = %s", dp_dev_info->branch_id_str);
+
+	/* set the corresponding AUX-ISP control library for the chip */
+	fu_kinetic_dp_device_set_aux_isp_ctrl(self, dp_dev_info->chip_id);
+
+	/* read current firmware version */
+	if (dp_dev_info->chip_id == KT_CHIP_JAGUAR_5000 ||
+	    dp_dev_info->chip_id == KT_CHIP_MUSTANG_5200) {
+		if (!fu_kinetic_dp_connection_read(connection,
+						   DPCD_ADDR_BRANCH_FW_MAJ_REV,
+						   buf_ver,
+						   DPCD_SIZE_BRANCH_FW_MAJ_REV +
+						       DPCD_SIZE_BRANCH_FW_MIN_REV +
+						       DPCD_SIZE_BRANCH_FW_REV,
+						   error))
+			return FALSE;
+	} else if (dp_dev_info->chip_id == KT_CHIP_PUMA_2900) {
+		/* read major and minor version */
+		if (!fu_kinetic_dp_connection_read(connection,
+						   DPCD_ADDR_BRANCH_FW_MAJ_REV,
+						   buf_ver,
+						   DPCD_SIZE_BRANCH_FW_MAJ_REV +
+						       DPCD_SIZE_BRANCH_FW_MIN_REV,
+						   error))
+			return FALSE;
+
+		/* read sub */
+		if (!fu_kinetic_dp_connection_read(connection,
+						   DPCD_ADDR_BRANCH_FW_SUB,
+						   &(buf_ver[2]),
+						   DPCD_SIZE_BRANCH_FW_SUB,
+						   error))
+			return FALSE;
+	}
+
+	version = g_strdup_printf("%1d.%03d.%02d", buf_ver[0], buf_ver[1], buf_ver[2]);
+	g_debug("device current firmware version %s", version);
+	fu_device_set_version(FU_DEVICE(self), version);
+
+	/* set up the device name */
+	chip_id = fu_kinetic_dp_aux_isp_get_chip_id_str(dp_dev_info->chip_id);
+	fu_device_set_name(FU_DEVICE(self), chip_id);
+
+	/* detect chip family */
+	self->family = fu_kinetic_dp_chip_id_to_family(dp_dev_info->chip_id);
+	switch (self->family) {
+	case FU_KINETIC_DP_FAMILY_JAGUAR:
+		/* TODO: set max firmware size for Jaguar */
+		g_debug("device family is Jaguar");
+		break;
+	case FU_KINETIC_DP_FAMILY_MUSTANG:
+		/* TODO: determine max firmware size for Mustang */
+		g_debug("device family is Mustang");
+		break;
+	case FU_KINETIC_DP_FAMILY_PUMA:
+		/* TODO: determine max firmware size for Puma */
+		g_debug("device family is Puma");
```

Comment:
the device family (and mode etc) needs to go in a FuDevice->to_string() function.

Reply:
and not have a g_debug here

Reply:
fixed